lib/tasks/valid_test_data_generator/base_populater.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require "active_support/testing/time_helpers"
+
+module ValidTestDataGenerator
+  class BasePopulater
+    include ActiveSupport::Testing::TimeHelpers
+
+    class << self
+      def populate(name:, total_schools: 10, participants_per_school: 10, cohort: Cohort.current, number_of_participants: 100)
+        new(name:, total_schools:, participants_per_school:, cohort:, number_of_participants:).populate
+      end
+    end
+
+    def populate
+      raise "Not implemented"
+    end
+
+  private
+
+    attr_reader :lead_provider, :total_schools, :participants_per_school, :cohort, :number_of_participants, :npq_courses, :logger
+
+    def initialize(name:, total_schools:, participants_per_school:, cohort:, number_of_participants:, logger: Rails.logger)
+      @lead_provider = ::NPQLeadProvider.find_or_create_by!(name:)
+      @total_schools = total_schools
+      @participants_per_school = participants_per_school
+      @cohort = cohort
+      @number_of_participants = number_of_participants
+      @logger = logger
+      # Ignoring ASO course, is an old course which we shouldn't create data
+      @npq_courses = NPQCourse.all.reject { |c| c.identifier == "npq-additional-support-offer" }
+    end
+
+    def create_application!(lead_provider:, school:, npq_course:, cohort:, participant_identity:)
+      NPQApplication.create!(
+        active_alert: "",
+        date_of_birth: Date.new(1990, 1, 1),
```

Comment:
is this because of the magic dates? or could we change this to be more random, if it's because of magic dates we can keep the same

Reply:
Yeah probably magic dates related. 👍